bundles/org.openhab.binding.linuxinput/src/main/java/org/openhab/binding/linuxinput/internal/LinuxInputDiscoveryService.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linuxinput.internal;
+
+import static org.openhab.binding.linuxinput.internal.LinuxInputBindingConstants.THING_TYPE_DEVICE;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.linuxinput.internal.evdev4j.EvdevDevice;
+import org.openhab.binding.linuxinput.internal.evdev4j.LastErrorException;
+import org.openhab.binding.linuxinput.internal.evdev4j.jnr.EvdevLibrary;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for LinuxInputHandlers based on the /dev/input directory.
+ *
+ * @author Thomas Weißschuh - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.linuxinput")
+public class LinuxInputDiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(LinuxInputDiscoveryService.class);
+
+    private Future<?> discoveryJob;
+    private static final Duration REFRESH_INTERVAL = Duration.ofSeconds(50);
```

Comment:
Personally I prefer `Duration`s over units in variable names.
Is this a project-wide preference or your own?

Reply:
It's a project-wide preference to have the unit in the variable name. Here you have a duration so it's more clear from the context. But in general int (or long) is used and then people tend to write in comment what the unit is. So therefor it's recommended to use the unit in the name. But I guess when using duration it's not really needed to specify this because it's just what you make it. So I guess your usage is ok.

Reply:
Regarding usage of `Duration` there is no project-wide decision or opinion. It was introduced with java 8 and the project originally (at least when I joined) was using Java 7.

Performance wise it costs, but it is not used in a hot path, so I'd say using `Duration` together with this current constant name is fine.